app/scripts/controllers/transactions/index.js

```diff
@@ -2078,19 +2080,41 @@ export default class TransactionController extends EventEmitter {
       eip1559Version = eip1559V2Enabled ? '2' : '1';
     }
 
+    const contractInteractionTypes = [
+      TRANSACTION_TYPES.CONTRACT_INTERACTION,
+      TRANSACTION_TYPES.TOKEN_METHOD_APPROVE,
+      TRANSACTION_TYPES.TOKEN_METHOD_SAFE_TRANSFER_FROM,
+      TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER,
+      TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER_FROM,
+    ].includes(type);
+
+    let transactionType;
+    if (type === TRANSACTION_TYPES.CANCEL) {
+      transactionType = TRANSACTION_TYPES.CANCEL;
+    } else if (type === TRANSACTION_TYPES.SIMPLE_SEND) {
+      transactionType = TRANSACTION_TYPES.SIMPLE_SEND;
+    } else if (contractInteractionTypes) {
+      transactionType = TRANSACTION_TYPES.CONTRACT_INTERACTION;
+    }
+
+    let transactionContractMethod;
+    if (transactionType === TRANSACTION_TYPES.CONTRACT_INTERACTION) {
+      transactionContractMethod = transactions[id]?.contractMethodName;
+    }
```

Comment:
@darkwing pointed out to me that we aren't caring for all types here, and transactionType will be undefined in the following cases with this code:

SWAP
SWAP_APPROVAL
SMART
TOKEN_METHOD_SET_APPROVAL_FOR_ALL
RETRY
DEPLOY_CONTRACT

are these meant to be stripped from metrics? @danjm @bschorchit

Reply:
I know there is a separate PR for the speed up, but the way this code is wirtten if a retry comes through in this state it'll have no 'transaction_type'

Reply:
Thaanks Brad and David for catching that! Here's how we're addressing it:

- SWAP, SWAP_APPROVAL, SMART, TOKEN_METHOD_SET_APPROVAL_FOR_ALL will now have transaction_type = contract_interaction

- DEPLOY_CONTRACT will now have transaction_type = contract_deployment

- RETRY will have transaction_type = the original transaction type which we're doing in this separate PR [here](https://github.com/MetaMask/metamask-extension/pull/15197)

Reply:
Thank you @brad-decker @darkwing for catching that! I made changes [here](https://github.com/MetaMask/metamask-extension/pull/15175/commits/d4c7e4d5eda07230006b494dc6ef92a38ba34970) and added missing transaction types like @bschorchit mentioned above.